### DISCUSSION TITLE
怎么下载qq群文件到本地

### DISCUSSION THREAD

**Discussion by lmlybmh** [simple-robot/Discussions#187] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

怎么可以将群成员上传的文件下载到本地

**Comment by ForteScarlet** [comment]
在 [MiraiGroup](https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai/-mirai-group/index.html) 中获取到 [originalContact](https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai/-mirai-group/index.html#-1365423508%2FProperties%2F1961516927)，也就是原始的 mirai 中的 `Group` 类型，它其中有可以操作群文件的 `files` 属性。

**Reply by lmlybmh** [reply]
不好意思我是刚接触这个，这个MiraiGroup要去怎么获取

**Reply by ForteScarlet** [reply]
mirai组件中的 `Bot` 的实际类型都是 `MiraiBot`，使用它就能获取到 `MiraiGroup`；
或者监听mirai组件下更具体的某个事件类型，例如 `MiraiGroupMessageEvent`，也能直接获取 `MiraiGroup`；
或者在拿到一个 `Group` 之后，判断其类型 ` if (group instanceof MiraiGroup miraiGroup) { ... }` 然后进行操作，都可以
